spec/lib/pre_assembly/from_staging_location/structural_builder_spec.rb

```diff
@@ -11,70 +11,116 @@
+
+          # It retains the collection
+          expect(structural.isMemberOf).to eq ['druid:bb000kk0000']
+        end
       end
-    end
 
-    context 'with world access' do
-      let(:dro_access) { { view: 'world' } }
-      let(:all_files_public) { false }
+      context 'with world access' do
+        let(:dro_access) { { view: 'world' } }
+        let(:all_files_public) { false }
 
-      it 'adds all the files' do
-        file_sets = structural.contains
-        expect(file_sets.size).to eq 1
-        files = file_sets.flat_map { |file_set| file_set.structural.contains }
-        expect(files.map(&:filename)).to eq ['document.pdf']
-        expected_access = { view: 'world', download: 'none', controlledDigitalLending: false }
-        expect(files.map(&:access).map(&:to_h)).to all(eq(expected_access))
+        it 'adds all the files' do
+          file_sets = structural.contains
+          expect(file_sets.size).to eq 1
+          files = file_sets.flat_map { |file_set| file_set.structural.contains }
+          expect(files.map(&:filename)).to eq ['document.pdf']
+          expected_access = { view: 'world', download: 'none', controlledDigitalLending: false }
+          expect(files.map(&:access).map(&:to_h)).to all(eq(expected_access))
 
-        # it stores administrative settings based on the file_attributes
-        expect(files.map { |file| file.administrative.to_h }).to all(eq({ publish: true, shelve: false, sdrPreserve: true }))
+          # it stores administrative settings based on the file_attributes
+          expect(files.map { |file| file.administrative.to_h }).to all(eq({ publish: true, shelve: false, sdrPreserve: true }))
 
-        # It retains the collection
-        expect(structural.isMemberOf).to eq ['druid:bb000kk0000']
+          # It retains the collection
+          expect(structural.isMemberOf).to eq ['druid:bb000kk0000']
+        end
+      end
+
+      context 'with dark access' do
+        let(:dro_access) { { view: 'dark', download: 'none' } }
+        let(:all_files_public) { false }
+
+        it 'adds all the files' do
+          file_sets = structural.contains
+          expect(file_sets.size).to eq 1
+          files = file_sets.flat_map { |file_set| file_set.structural.contains }
+          expect(files.map(&:filename)).to eq ['document.pdf']
+          expected_access = { view: 'dark', download: 'none', controlledDigitalLending: false }
+          expect(files.map(&:access).map(&:to_h)).to all(eq(expected_access))
+
+          # it stores administrative settings corresponding to the access
+          expect(files.map { |file| file.administrative.to_h }).to all(eq({ publish: false, shelve: false, sdrPreserve: true }))
+
+          # It retains the collection
+          expect(structural.isMemberOf).to eq ['druid:bb000kk0000']
+        end
       end
     end
 
-    context 'with dark access' do
-      let(:dro_access) { { view: 'dark', download: 'none' } }
-      let(:all_files_public) { false }
```

Comment:
note: most of the this spec didn't change, i just moved a lot of in into a new `context 'with flat file structure'` block and then added a new `context 'with hierarchical file structure'` below